src/main/java/com/microsoft/sqlserver/jdbc/SQLServerXAResource.java

```diff
@@ -211,73 +211,89 @@ public String toString() {
-                break;
-            case SQLServerXAResource.XA_PREPARE_EX:
-                CS = controlConnection.prepareCall("{call master..xp_sqljdbc_xa_prepare_ex(?, ?, ?, ?, ?, ?)}");
-                break;
-            case SQLServerXAResource.XA_ROLLBACK_EX:
-                CS = controlConnection.prepareCall("{call master..xp_sqljdbc_xa_rollback_ex(?, ?, ?, ?, ?, ?)}");
-                break;
-            case SQLServerXAResource.XA_FORGET_EX:
-                CS = controlConnection.prepareCall("{call master..xp_sqljdbc_xa_forget_ex(?, ?, ?, ?, ?, ?)}");
-                break;
-            default:
-                assert false : "Bad handle request:" + number;
-                break;
-        }
+            CallableStatement CS = null;
+
+            switch (number) {
+                case SQLServerXAResource.XA_START:
+                    CS = controlConnection.prepareCall(
+                            "{call master..xp_sqljdbc_xa_start(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)}");
+                    break;
+                case SQLServerXAResource.XA_END:
+                    CS = controlConnection.prepareCall("{call master..xp_sqljdbc_xa_end(?, ?, ?, ?, ?, ?, ?)}");
+                    break;
+                case SQLServerXAResource.XA_PREPARE:
+                    CS = controlConnection.prepareCall("{call master..xp_sqljdbc_xa_prepare(?, ?, ?, ?, ?)}");
+                    break;
+                case SQLServerXAResource.XA_COMMIT:
+                    CS = controlConnection.prepareCall("{call master..xp_sqljdbc_xa_commit(?, ?, ?, ?, ?, ?)}");
+                    break;
+                case SQLServerXAResource.XA_ROLLBACK:
+                    CS = controlConnection.prepareCall("{call master..xp_sqljdbc_xa_rollback(?, ?, ?, ?, ?)}");
+                    break;
+                case SQLServerXAResource.XA_FORGET:
+                    CS = controlConnection.prepareCall("{call master..xp_sqljdbc_xa_forget(?, ?, ?, ?, ?)}");
+                    break;
+                case SQLServerXAResource.XA_RECOVER:
+                    CS = controlConnection.prepareCall("{call master..xp_sqljdbc_xa_recover(?, ?, ?, ?)}");
+                    break;
+                case SQLServerXAResource.XA_PREPARE_EX:
+                    CS = controlConnection.prepareCall("{call master..xp_sqljdbc_xa_prepare_ex(?, ?, ?, ?, ?, ?)}");
+                    break;
+                case SQLServerXAResource.XA_ROLLBACK_EX:
+                    CS = controlConnection.prepareCall("{call master..xp_sqljdbc_xa_rollback_ex(?, ?, ?, ?, ?, ?)}");
+                    break;
+                case SQLServerXAResource.XA_FORGET_EX:
+                    CS = controlConnection.prepareCall("{call master..xp_sqljdbc_xa_forget_ex(?, ?, ?, ?, ?, ?)}");
+                    break;
+                default:
+                    assert false : "Bad handle request:" + number;
+                    break;
+            }
```

Comment:
are these tabs? please make sure to use the code [formatter](https://github.com/microsoft/mssql-jdbc/blob/main/mssql-jdbc_formatter.xml)

Reply:
I'm not seeing tabs and am using the mssql-jdbc formatter.  AFAIK, I've never used tabs for any coding.

Reply:
GitHub makes it hard to tell if they are space or tabs, thanks for confirming that these are spaces.